### PR TITLE
[AB#16670] add ErrorBoundary to top level app to catch render exception

### PR DESCRIPTION
**Pull request by mydiemho**
also add an example on how to use the global error redux
for caught exception.

### CODE REVIEW

File: src/react/components/pages/homepage/homePage.tsx [hunk 1]
```diff
@@ -116,8 +121,20 @@ export default class HomePage extends React.Component<IHomepageProps> {
     }
 
     private onProjectFileUpload = (e, projectJson) => {
```
Comment (wbreza) [inline src/react/components/pages/homepage/homePage.tsx]: I would like to see a test for this scenario to verify your showError is called when the user attempt to upload a invalid file.

File: src/react/components/pages/homepage/homePage.tsx [hunk 2]
```diff
@@ -115,9 +118,20 @@ export default class HomePage extends React.Component<IHomepageProps> {
...
-    private onProjectFileUpload = async (e, projectJson) => {
-        const project: IProject = JSON.parse(projectJson);
-        await this.loadSelectedProject(project);
+    private onProjectFileUpload = async (e, project) => {
+        let projectJson: IProject;
+        try {
+            projectJson = JSON.parse(project);
+        } catch (error) {
+            this.props.appErrorActions.showError({
+                title: "Homepage has an error",
```
Comment (wbreza) [inline src/react/components/pages/homepage/homePage.tsx]: Nit: we should start localizing our known exception messages.

File: src/react/components/pages/homepage/homePage.tsx [hunk 3]
```diff
@@ -115,9 +118,20 @@ export default class HomePage extends React.Component<IHomepageProps> {
...
+                message: "File is not valid json",
+            });
+
+            return;
+        }
+
+        await this.loadSelectedProject(projectJson);
```
Comment (wbreza) [inline src/react/components/pages/homepage/homePage.tsx]: You can move this `await this.loadSelectedProject(projectJson);` into the `try` block which will then avoid you having to have the return statement in your `catch` block

Comment (mydiemho) [reply src/react/components/pages/homepage/homePage.tsx]: that mean we could have an error that is coming from loadSelectedProject and not just from json parsing, which is fine but the error message would have to be more generic

**Pull request merged**